src/Compilers/Core/Portable/NativePdbWriter/ISymUnmanagedWriter.cs

```diff
@@ -147,48 +147,107 @@ internal interface IPdbWriter
     [StructLayout(LayoutKind.Explicit)]
     internal struct VariantStructure
     {
-        public VariantStructure(DateTime date) : this() // Need this to avoid errors about the uninteresting union fields.
+        public unsafe VariantStructure(DateTime date) : this() // Need this to avoid errors about the uninteresting union fields.
         {
-            _longValue = date.Ticks;
+            fixed (VariantStructure* thisPtr = &this)
+            {
+                *((long*)((byte*)thisPtr+8)) = date.Ticks;
+            }
+
             _type = (short)VarEnum.VT_DATE;
         }
 
         [FieldOffset(0)]
         private readonly short _type;
 
-        [FieldOffset(8)]
-        private readonly long _longValue;
-
```

Comment:
I didn't find any such bug. I will file a bug on .NET Native side to track this.